flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerGateway.java

```diff
@@ -104,6 +106,7 @@
 		ResourceID resourceId,
 		int dataPort,
 		HardwareDescription hardwareDescription,
+		ResourceProfile defaultSlotResourceProfile,
```

Comment:
Looks like we keep exploding arguments for this method. It would be nice to add a hotfix commit with a refactoring where this method accepts a class container for those parameters, like `TaskExecutorByResourceManagerRegistration` or something.

Reply:
Not sure if we want to reuse `WorkerRegistration` here.
The functionality seems quite similar, but I'm not sure whether we want to expose `WorkerRegistration` to outside of resource manager.

Reply:
we do not necessary need to reuse `WorkerRegistration`, it could be a separate API class container of these method parameters.

Reply:
I'm in favor of naming the class container `ResourceManagerGatewayRegisterTaskExecutorParams`, so that we avoid to confuse people with the name of `xxxRegistration` that similar to `WorkerRegistration`.

Reply:
As discussed offline, we decided to call this class `TaskExecutorRegistration` to avoid too long class names.